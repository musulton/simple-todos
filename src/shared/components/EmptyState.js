import {Text, View, StyleSheet} from "react-native";

const EmptyState = ({text}) => {
    return (
        <View style={styles.container}>
            <Text style={styles.text}>{text}</Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
    },
    text: {
        fontSize: 16,
        fontWeight: "bold"
    }
})

export default EmptyState
