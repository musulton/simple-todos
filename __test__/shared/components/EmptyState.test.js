import {render, screen} from "@testing-library/react-native";

import EmptyState from "../../../src/shared/components/EmptyState";

describe("EmptyState", () => {
    test("should render correctly", () => {
        const textProps = "Data is empty"
        
        render(<EmptyState text={textProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
