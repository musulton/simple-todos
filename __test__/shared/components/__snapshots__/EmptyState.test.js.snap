// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Data is empty
  </Text>
</View>
`;
